components/service/glean/src/main/java/mozilla/components/service/glean/storages/EventsStorageEngine.kt

```diff
@@ -0,0 +1,83 @@
+package mozilla.components.service.glean.storages
+
+import android.os.SystemClock
+import android.support.annotation.VisibleForTesting
+
+/**
+ * This singleton handles the in-memory storage logic for events. It is meant to be used by
+ * the Specific Events API and the ping assembling objects. No validation on the stored data
+ * is performed at this point: validation must be performed by the Specific Events API.
+ */
+object EventsStorageEngine {
```

Comment:
After discussing this with Sebastian, looks like we can simply make the storage `internal` so that it's only accessible from within the library. Pretty neat!